src/common/host_scalar_memory_storage.hpp

```diff
@@ -45,20 +47,34 @@ class host_scalar_memory_storage_t : public memory_storage_t {
 
     status_t get_scalar_value(void *value, size_t value_size) const {
         if (!is_initialized_ || value == nullptr
-                || value_size != types::data_type_size(data_type_))
+                || value_size != types::data_type_size(data_type_)
+                || value_size == 0 || value_size > max_scalar_size)
             return status::invalid_arguments;
 
-        std::memcpy(value, &scalar_storage_, value_size);
+        // Use std::copy with explicit byte pointers to help compiler verify bounds
```

Comment:
I think `std::copy` cannot verify the bounds in this case but using it will force the compiler to avoid using `__builtin_memcpy` that triggers the the warning. 

Ideally, I think we should ignore the warning because it's GCC's buggy behavior. But if we don't want to deal with disabling the warning for this part of the case we can keep this fix as is and leave a comment that this `std::copy` has to stay here so that we don't remove it accidently in the future.